onchain/example-program-on-bonsol/src/lib.rs

```
@@ -0,0 +1,155 @@
+                ExecutionConfig {
+                    verify_input_hash: true,
+                    input_hash: Some(input_hash),
+                    forward_output: true,
+                },
+                Some(CallbackConfig {
+                    program_id: crate::id(),
+                    instruction_prefix: vec![1],
+                    extra_accounts: vec![
+                        AccountMeta::new(*requester.key, false),
+                        AccountMeta::new(EA1, false),
+                        AccountMeta::new_readonly(EA2, false),
+                        AccountMeta::new_readonly(EA3, false),
+                    ],
+                }),
+            )
+            .map_err(|_| ProgramError::InvalidInstructionData)?;
+            invoke_signed(&ix, accounts, &[&[execution_id.as_bytes(), &[bump]]])?;
+            let mut data = requester.try_borrow_mut_data()?;
+            data.copy_from_slice(&execution_account.key.to_bytes());
+            Ok(())
+        }
+        1 => {
+            let requester = &accounts[1];
+            let requester_data = requester.try_borrow_data()?;
+            let execution_account = Pubkey::try_from(&requester_data[0..32])
+                .map_err(|_| ProgramError::InvalidInstructionData)?;
+            let callback_output: BonsolCallback =
+                handle_callback(SIMPLE_IMAGE_ID, &execution_account, accounts, data)?;
+            if sol_memcmp(accounts[2].key.as_ref(), EA1.as_ref(), 32) != 0 {
+                return Err(ProgramError::InvalidInstructionData.into());
+            }
+            if sol_memcmp(accounts[3].key.as_ref(), EA2.as_ref(), 32) != 0 {
+                return Err(ProgramError::InvalidInstructionData.into());
+            }
+            if sol_memcmp(accounts[4].key.as_ref(), EA3.as_ref(), 32) != 0 {
+                return Err(ProgramError::InvalidInstructionData.into());
+            }
+            assert!(accounts[2].is_writable, "Writable account not found");
```

this should return an error probably?